support/cas-server-support-trusted-mfa/src/main/java/org/apereo/cas/trusted/fingerprint/DeviceFingerprintGenerator.java

```diff
@@ -0,0 +1,23 @@
+package org.apereo.cas.trusted.fingerprint;
+
+import org.springframework.webflow.execution.RequestContext;
```

Comment:
I think we might want to rename the interface, if it's to deal with or support specific webflow concepts?

Reply:
Why does the interface need to depend on the SWF `RequestContext`? If it does, I'd move the definition of this API into `web.flow` package, otherwise, don't "leak" SWF types outside of that package.

Reply:
I'll update the name and move it over to the `web.flow` package.

I included the `RequestContext` object in order for implementors to have access to the request to extract meta-data about the request to generate the device fingerprint. In our local use case this will be replacing the pre-existing geography fingerprint with a cookie based solution.

The current geography implementation uses the `RequestContext` object, but accesses it through `RequestContextHolder`. I think passing in the RequestContext object directly is a better pattern when the object is already directly available.

Reply:
> I think passing in the RequestContext object directly is a better pattern when the object is already directly available.

Agreed.

Reply:
👍 To `web.flow` package then